client/client.go

```diff
@@ -289,10 +276,15 @@ func NewClient(svrAddrs []string, security SecurityOption, opts ...ClientOption)
 func NewClientWithContext(ctx context.Context, svrAddrs []string, security SecurityOption, opts ...ClientOption) (Client, error) {
 	log.Info("[pd] create pd client with endpoints", zap.Strings("pd-address", svrAddrs))
 	c, clientCtx, clientCancel, tlsCfg := createClient(ctx, &security)
-	c.tsoStreamBuilderFactory = &pdTSOStreamBuilderFactory{}
-	c.bc = newPDBaseClient(clientCtx, clientCancel, &c.wg, addrsToUrls(svrAddrs), tlsCfg, c.option)
-	c.tsobc = c.bc
-	if err := c.setup(opts...); err != nil {
+	// Inject the client options.
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	c.pdsd = newPDServiceDiscovery(clientCtx, clientCancel, &c.wg, addrsToUrls(svrAddrs), tlsCfg, c.option)
+	c.tsoc = newTSOClient(clientCtx, clientCancel, &c.wg, c.option, c.keyspaceID, c.pdsd, c.pdsd.(tsoAllocatorEventSource), &pdTSOStreamBuilderFactory{})
```

Comment:
In this situation, the keyspace is 0, am I right?

Reply:
It's 0 for now, but it needs to be changed after we discuss with the serverless tier. 0 has already been used by the existing tenants. The max number of tenant per cluster MAX_VALUE is limited to 0xFFFFFF. We need a value > MAX_VALUE to represent non-tenant-awareness users. I rememeber someone in serverless team mentioned they already used such a value and we might need to reuse it or use 0x1000000.